demo.md

```diff
@@ -0,0 +1,155 @@
+# **RVIP** [Out The Box]
+**[Version]:** 1.0.0
+
+# **What Is This?**
+This document provides instruction to set-up **RVIP** as a complete Robotic Vision suite.
+
+This is based on the combination of ROS packages and external libraries used for testing purposes.
+
+This **suite** is made of:
+1. OpenGR
+2. darknet_ros
+3. rvip
+
+# **Dependencies**
+### This section lists all third-party softwares used by **RVIP**.
+
+**Click** the hyperlinks for download instructions.
+
+1. [Ubuntu 18.04 (Bionic Beaver)](https://ubuntu.com/download/desktop)
+    - For the operating system
+2. [ROS Morenia Melodic](http://wiki.ros.org/melodic/Installation/Ubuntu)
+    - For the ROS base
+3. OpenGR
+    - For pose estimation. `Refer below for download instructions.`
+4. [OpenCV 3.2.0](http://www.codebind.com/linux-tutorials/install-opencv-3-2-ubuntu-18-04-lts-linux/)
+    - For reading and displaying images
+5. [Point Cloud Library](https://askubuntu.com/questions/1134856/how-to-install-pcl-on-ubuntu-18-04)
+    - For processing 3D information
+6. darknet_ros
+    - For object recognition using only 2D RGB images. `Refer below for download instructions.`
```

Comment:
This section is somewhat confusing: do users need to install all of this themselves? Are parts installed automatically by either the build scripts or later sections in these instructions?

I would suggest to not link to either the Ubuntu or the ROS setup instructions. This is a ROS package, we can certainly state compatibility with ROS Melodic (on Ubuntu Bionic), but users should then be able to figure things out for themselves).

Reply:
> do users need to install all of this themselves?

Technically speaking, yes. Users need to install all of this themselves. From the perspective of my target audience (people with low to no programming background), I believe it is necessary to specify all required software components that come together to make **rvip** work.

> Are parts installed automatically by either the build scripts or later sections in these instructions?

The specified parts under this section **Requirements** will not be automatically downloaded and built. Users will have to click on the hyperlinks and download themselves.

 > I would suggest to not link to either the Ubuntu or the ROS setup instructions.

The hyperlinks should remain and serve as a convenience for users who need to go through the prior installation setup of software components under **Requirements**.

Do let me know if you think it is okay to leave it as it is, based on the reasons I have given. @gavanderhoorn

Reply:
This may connect to the "novices vs experienced users" discussion.

:+1: for detailed documentation, but it would be good to think about which is your default target audience: novices or people with (a bit) more experience.

I would suggest to split these steps into two: one with all the detail you have now, and one which just lists required versions of dependencies like Ubuntu and ROS. More experienced users should either already have those, or be capable of installing them.

For dependencies with special requirements (such as OpenGR), it is of course OK to be verbose.

Reply:
Addressed in [this commit](https://github.com/ros-industrial/rvip/pull/1/commits/36d50eac3082a03d2f9b154c2fb30f1815660cb4).